tensorzero-core/src/inference/types/extra_body.rs

```diff
@@ -69,30 +78,425 @@ pub struct FullExtraBodyConfig {
     pub inference_extra_body: FilteredInferenceExtraBody,
 }
 
-#[derive(Clone, Debug, Deserialize, PartialEq, Serialize, ts_rs::TS)]
-#[serde(untagged)]
-pub enum InferenceExtraBody {
-    Provider {
-        model_provider_name: String,
-        pointer: String,
-        #[serde(flatten)]
-        kind: ExtraBodyReplacementKind,
-    },
-    Variant {
-        variant_name: String,
-        pointer: String,
-        #[serde(flatten)]
-        kind: ExtraBodyReplacementKind,
-    },
+pub mod dynamic {
+    use schemars::JsonSchema;
+    use serde::{Deserialize, Serialize};
+    use tensorzero_derive::export_schema;
+
+    #[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize, ts_rs::TS)]
+    #[ts(optional_fields)]
+    #[export_schema]
+    #[serde(untagged, deny_unknown_fields)]
+    pub enum ExtraBody {
+        #[schemars(title = "ProviderExtraBody")]
+        #[deprecated(note = "Migrate to `ModelProvider` and remove in 2026.2+. (#4640)")]
+        /// DEPRECATED: Use `ModelProvider` instead.
+        Provider {
+            /// A fully-qualified model provider name in your configuration (e.g. `tensorzero::model_name::my_model::provider_name::my_provider`)
+            model_provider_name: String,
+            /// A JSON Pointer to the field to update (e.g. `/enable_agi`)
+            pointer: String,
+            /// The value to set the field to
+            value: serde_json::Value,
+        },
+        #[schemars(title = "ProviderExtraBodyDelete")]
+        #[deprecated(note = "Migrate to `ModelProviderDelete` and remove in 2026.2+. (#4640)")]
+        /// DEPRECATED: Use `ModelProviderDelete` instead.
+        ProviderDelete {
+            /// A fully-qualified model provider name in your configuration (e.g. `tensorzero::model_name::my_model::provider_name::my_provider`)
+            model_provider_name: String,
+            /// A JSON Pointer to the field to update (e.g. `/enable_agi`)
+            pointer: String,
+            #[serde(
+                serialize_with = "super::super::serialize_delete_field",
+                deserialize_with = "super::super::deserialize_delete_field"
+            )]
+            #[schemars(schema_with = "super::super::schema_for_delete_field")]
+            /// Set to true to remove the field from the model provider request's body
+            delete: (),
```

Comment:
Why not use `delete: bool` here? You could still have custom deserializer to validate it's `true`, but we could also just say "if it's `false` it has no effect" and it's not complicated to implement. Having unit tuple interact with JSON feels like asking for trouble since you can't cleanly represent a unit tuple in most other programming languages.